docs/roles-and-permissions.md

```diff
@@ -3,49 +3,66 @@ Roles and permissions in the application have been setup in a flexible and maint
 
-The application uses a dedicated convention in order to define and populate roles and permissions, leaving little in the way of thinking when it comes time to stand up the application locally or in a live environment. Roles and permissions are mainly controlled through the `authorization` module, but each additional module also has some responsibility in order to correctly configure what it may need to properly facilitate role based access control.
+The application uses a dedicated convention in order to define and populate roles and permissions, leaving little in the way of thinking when it comes time to stand up the application locally or in a live environment.
 
 ### Local Setup
-In order to get your local environment correctly set up, you won't have to do anything beyond running the `composer refresh-database` command. Under the hood, this command will call on the `DatabaseSeeder`, which calls the `SyncRolesAndPermissions` artisan command. This command will seed and sync all of the roles and permissions defined in the application.
+In order to get your local environment correctly set up, you won't have to do anything beyond running the setup process defined in the documentation. Permissions are created in data migrations and roles are created by seeders when a Tenant is created.
 
 ## Permissions
-The application defines two distinct types of permissions: those related directly to models and those that are not. In the application, these are aptly referred to as `model` permissions and `custom` permissions.
+Permissions in the application are created and managed through special data Migrations called Permission Migrations. Details on Data Migrations can be found in the [Data Migrations](/docs/data-migrations.md) documentation.
 
-The application defines a sane set of defaults for each model in the application, but also provides extensibility. In order to ensure that any new model introduced in the application automatically has the default `model` permissions defined for it, the following needs to be done:
+### Permission Migrations
+Permission Migrations operate much the same as regular Data Migrations but adhere to a few slightly different or additional rules:
 
-1. Register the model with the `Relation::morphMap()` in your module service provider
-2. Extend the `BaseModel`, or directly use the `AdvisingApp\Authorization\Models\Concerns\DefinesPermissions.php` trait on your model.
+1. Prepend the name of the migration with `seed_permissions_` to differentiate it from schema migrations and Data Migrations.
+2. The `Database\Migrations\Concerns\CanModifyPermissions` Trait and it's available helpers should be used.
+3. Any queries should be surrounded by both a database transaction and a `try/catch` block to catch any SQL errors such as a `UniqueConstraintViolation`. This is to prevent issues with permission duplication and the corruption of the current connection if such an error is triggered.
```

Comment:
```suggestion
3. Any other queries you write should be surrounded by both a database transaction and a `try`/`catch` block to catch any SQL errors such as a `UniqueConstraintViolation`.
```

Removing that second bit because its now handled by the `insertOrIgnore()`